app/models/concerns/check_elastic_search_model.rb

```diff
@@ -46,19 +46,15 @@ def self.get_index_name
     CONFIG['elasticsearch_index'].blank? ? [Rails.application.engine_name, Rails.env, 'annotations'].join('_') : CONFIG['elasticsearch_index']
   end
 
-  def self.reindex_es_data(mapping_keys = nil)
-    mapping_keys = [MediaSearch, CommentSearch, TagSearch, DynamicSearch] if mapping_keys.nil?
+  def self.reindex_es_data
     source_index = CheckElasticSearchModel.get_index_name
     target_index = "#{source_index}_reindex"
-    begin
-      MediaSearch.delete_index target_index
-      MediaSearch.migrate_es_data(source_index, target_index, mapping_keys)
-      sleep 1
-      MediaSearch.migrate_es_data(target_index, source_index, mapping_keys)
-      MediaSearch.index_name = source_index
-    rescue StandardError => e
-      Rails.logger.error "[ES MIGRATION] Could not start migation: #{e.message}"
-    end
+    # copy data to destination
+    MediaSearch.migrate_es_data(source_index, target_index)
+    sleep 20
```

Comment:
is there a way to ask ES if the new index has finished processing?

Reply:
Yes there is a response for reindex api, we can parse it